src/app/pages/tasks/component.ts

```diff
@@ -11,8 +11,21 @@ import { ApiService } from '@services/api';
 })
 export class TasksComponent implements OnInit {
   score = 0;
+  initialTasks: any[] = [];
   tasks: any[] = [];
   completedTasks: any[] = [];
+  taskTypeFilters: any = {
+    Completed: (task: any): boolean => {
+      return Boolean(task.completedOn);
+    },
+    Available: (task: any): boolean => {
+      return Boolean(task.active);
+    },
+    Unavailable: (task: any): boolean => {
+      return !Boolean(task.active);
+    },
+  };
```

Comment:
It's strange to see `Boolean` used in JavaScript; consider `!!` if you need strict boolean conversion, but most of the time even this is not necessary.

```suggestion
    completed: (task)   => task.completedOn,
    available: (task)   => task.active,
    unavailable: (task) => !task.active,
  };
```